src/analysis/inner_product_space/basic.lean

```diff
@@ -1066,6 +1066,43 @@ begin
   simp only [sq, ← mul_div_right_comm, ← add_div]
 end
 
+section complex
+
+variables (V : Type*)
+[inner_product_space ℂ V]
+/--
+A complex polarization identity, with a linear map
+-/
+lemma inner_map_polarization (T : V →ₗ[ℂ] V) (x y : V):
+  ⟪ T y, x ⟫_ℂ = (⟪T (x + y) , x + y⟫_ℂ - ⟪T (x - y) , x - y⟫_ℂ +
+    complex.I * ⟪T (x + complex.I • y) , x + complex.I • y⟫_ℂ -
+    complex.I * ⟪T (x - complex.I • y), x - complex.I • y ⟫_ℂ) / 4 :=
+begin
+  iterate {rw [map_add, inner_add_left, inner_add_right, inner_add_right]},
+  iterate {rw [map_sub,inner_sub_left, inner_sub_right, inner_sub_right]},
+  rw [linear_map.map_smul, inner_smul_left, inner_smul_right],
+  ring_nf,
+  rw complex.conj_I,
+  ring_nf,
+  rw complex.I_sq,
+  ring_nf,
```

Comment:
```suggestion
  simp only [map_add, map_sub, inner_add_left, inner_add_right, linear_map.map_smul,
             inner_smul_left, inner_smul_right, complex.conj_I, ←pow_two, complex.I_sq,
             inner_sub_left, inner_sub_right, mul_add, ←mul_assoc, mul_neg, neg_neg,
             sub_neg_eq_add, one_mul, neg_one_mul, mul_sub, sub_sub],
  ring,
```
That's the best I could do with this one. (Basically, what `simp` does is to try to rewrite as much as it can with the lemmas it is given, so when you have something like `rw` under an iterate block, it's probably a good idea to try `simp only` on that set of lemmas and see where that takes you.)

Reply:
I see, I wasn't sure if it was good practice to always use `simp` when possible (it seems like yes for `simp only`, no for blind simps).  Thanks for this!